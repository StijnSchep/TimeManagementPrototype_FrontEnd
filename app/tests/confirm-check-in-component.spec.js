import { shallowMount, createLocalVue } from '@vue/test-utils';
import ConfirmCheckIn from '../components/Check-in/ConfirmationCheckInComponent';
import Vuex from "vuex";
import CheckIn from "../components/Check-in/CheckInComponent";

const localVue = createLocalVue();
localVue.use(Vuex);

jest.mock("tns-core-modules/ui/dialogs", () => ({
  alert: jest.fn(() => Promise.resolve())
}));

describe('ConfirmCheckInComponent.vue', () => {
  let store;
  let state;
  let actions;

  beforeEach(() => {
    state = {
      workedHours: [],
      branchId: null,
      departmentId: null
    };
    actions = {
      getBranchIdAndDepartmentId: jest.fn(),
      getAllBranchesAndDepartments: jest.fn((c, d) => Promise.resolve({ data: {} })),
      postCheckIn: jest.fn((c, d) => Promise.resolve({ data: {} })),
      fetchAllWorkedHours: jest.fn()
    };
    store = new Vuex.Store({
      modules: {
        checkIn: {
          state,
          actions
        }
      }
    });
  });
  
  
  it('mounts and renders', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.html()).toBeTruthy();
  });

  it('renders correctly', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.element).toMatchSnapshot();
  });

  it('is a Vue instance', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('Expect component to exist', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper).toBeTruthy();
  });

  it('Expect component to have Page element', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.find('Page')).toBeTruthy();
  });

  it('Expect component to have StackLayout element', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.find('StackLayout')).toBeTruthy();
  });

  it('Expect component to have template element', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.find('template')).toBeTruthy();
  });

  it('Expect component to have script element', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.find('script')).toBeTruthy();
  });

  it('Expect component to have style element', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.find('style')).toBeTruthy();
  });

  it('Expect component to have Label element', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.find('Label')).toBeTruthy();
  });

  it('Expect component to have HoursPage Component', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(wrapper.find('HoursPage')).toBeTruthy();
  });

  it('Expect component to have props array property', async () => {
    const wrapper = shallowMount(ConfirmCheckIn, { store, localVue });
    await expect(typeof wrapper.props).toBe('function')
  });
});
