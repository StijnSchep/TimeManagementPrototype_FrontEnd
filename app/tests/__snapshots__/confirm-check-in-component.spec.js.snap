// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmCheckInComponent.vue renders correctly 1`] = `
<scrollview-stub>
  <stacklayout-stub
    orientation="vertical"
  >
    <cardview-stub
      class="card"
      elevation="10"
      radius="5"
    >
      <gridlayout-stub
        class="form"
        rows="auto, auto"
      >
        <gridlayout-stub
          row="0"
        >
          <label
            class="fa"
            id="checkIcon"
            text="fa-check-circle"
          />
        </gridlayout-stub>
         
        <gridlayout-stub
          row="1"
        >
          <label
            class="checkInText"
            text="Je bent om undefined ingeklokt"
          />
        </gridlayout-stub>
      </gridlayout-stub>
    </cardview-stub>
  </stacklayout-stub>
</scrollview-stub>
`;
